sagemaker_processing/local_pyspark/code/preprocess.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import csv
+import os
+import shutil
+import sys
+import time
+
+import pyspark
+from pyspark.sql import SparkSession
+from pyspark.ml import Pipeline
+from pyspark.ml.feature import (
+    OneHotEncoder,
+    StringIndexer,
+    VectorAssembler,
+    VectorIndexer,
+)
+from pyspark.sql.functions import *
+from pyspark.sql.types import (
+    DoubleType,
+    StringType,
+    StructField,
+    StructType,
+)
+
+
+def csv_line(data):
+    r = ",".join(str(d) for d in data[1])
+    return str(data[0]) + "," + r
+
+
+def main():
+    parser = argparse.ArgumentParser(description="app inputs and outputs")
+    parser.add_argument("--s3_input_bucket", type=str, help="s3 input bucket")
+    parser.add_argument("--s3_input_key_prefix", type=str, help="s3 input key prefix")
+    parser.add_argument("--s3_output_bucket", type=str, help="s3 output bucket")
+    parser.add_argument("--s3_output_key_prefix", type=str, help="s3 output key prefix")
+    args = parser.parse_args()
+
+    spark = SparkSession.builder.appName("PySparkApp").getOrCreate()
+
+    # This is needed to save RDDs which is the only way to write nested Dataframes into CSV format
+    spark.sparkContext._jsc.hadoopConfiguration().set(
+        "mapred.output.committer.class", "org.apache.hadoop.mapred.FileOutputCommitter"
+    )
```

Comment:
Understand this is prob legacy code, but don't want to build this into the example if it's not need. Why do we have a "Nested Dataframe" ?

Reply:
removed, updated code to write to csv using data frames.